### PR TITLE
ISPN-5222 Support combined filter/converter in remote listeners

### PR DESCRIPTION
**Pull request by galderz**
* Allow CacheEventFilterConverter instances to be plugged.
* No need to change Hot Rod protocol, simply assume that if both the filter and converter factories are the same, then a CacheEventFilterConverterFactory instance needs to be looked up, which provides instances of CacheEventFilterConverter.

### CODE REVIEW

**Comment by anistor** [comment]
Looking ...

File: server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/ExtensionManagerService.java [hunk 1]
```diff
@@ -21,6 +22,7 @@
     private final List<HotRodServer> servers = new ArrayList<>();
     private final Map<String, CacheEventFilterFactory> filterFactories = new HashMap<>();
     private final Map<String, CacheEventConverterFactory> converterFactories = new HashMap<>();
+    private final Map<String, CacheEventFilterConverterFactory> filterConverterFactories = new HashMap<>();
```
Comment (anistor) [inline server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/subsystem/ExtensionManagerService.java]: ```filterConverterFactories``` is not completely handled in the other addXXX methods.

**Comment by anistor** [comment]
RemoteListenerWithDslFilterTest fails. It seems JPACacheEventFilterConverterFactory needs some loving care. I'll add a commit to your PR in a few mins to fix this.

**Comment by anistor** [comment]
@galderz I added the fix for RemoteListenerWithDslFilterTest here: https://github.com/anistor/infinispan/tree/galderz_t_5222.

**Comment by anistor** [comment]
Looks nice. I'll integrate it once the two noted issues are fixed.

**Comment by galderz** [comment]
@anistor Updated and added your fix, thanks for the help :)

**Comment by galderz** [comment]
I also added better error reporting for users that might have been using filter/converters in the way `RemoteListenerWithDslFilterTest` did.

**Comment by anistor** [comment]
Looking ...

**Comment by anistor** [comment]
Integrated. Thanks @galderz !

**Pull request merged**